components/giftbox.vue

```diff
@@ -76,27 +76,72 @@
       loading: true,
       features: [],
       selectedFeature: undefined,
-      seenFeatures: new Set(),
     }),
+    computed: {
+      seenFeatures() {
+        return this.$cookies.get('seenFeatures') || {};
+      },
+      cookieOptions() {
+        return {
+          maxAge: 60 * 60 * 24 * 30,
+        };
+      },
+    },
     created() {
       this.fetchFeatures().then((res) => {
         this.features = res;
+        // sync the seenFeatures with the incomming features from the api call
+        this.syncSeenFeatures();
+        // if there are new features
         if (this.features.length > 0) {
           this.selectedFeature = this.features[0];
-          this.seenFeatures.add(this.selectedFeature.uid);
+          this.addToSeenFeatures(this.selectedFeature.uid);
         }
       });
     },
     methods: {
       async fetchFeatures() {
         this.loading = true;
-        const response = await fetch(
-          process.env.newFeaturesUrl
-        );
+        const response = await fetch(process.env.newFeaturesUrl);
         const { data } = await response.json();
         this.loading = false;
         return data;
       },
+      addToSeenFeatures(uid) {
+        const seenFeatures = this.seenFeatures;
+        seenFeatures[uid] = true;
```

Comment:
Why not make it an array of uids, e.g. `[uid1, uid2, etc]` instead of `{uid1: true, uid2: true, etc}`?

If it's just to easily check if the uid is in the array, you can use [Array.prototype.includes](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Array/includes)

To remove a uid from the array you could use [Array.prototype.filter](https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

I feel like this would make the logic a bit easier to reason about when someone has to re-read this in the future to make changes.